### ISSUE TITLE
Expose specific ports to the web whilst VPN is active

### ISSUE THREAD

**Issue by fad6b5f049** [dperson/openvpn-client#202]
Hi!

After the firewall change the port-forward (used for incoming connections) from my VPN provider is no longer working.

Is there a way to allow specific port-forwards whilst VPN tunnel is active with the new change, or is this out-of-scope for this container?

**Comment by fad6b5f049** [comment]
I've tried the -p "<PORT>" described in another issue (https://github.com/dperson/openvpn-client/issues/201) as well as -e VPNPORT=<PORT> with neither of them working.

**Comment by claylong** [comment]
Having the same issue, working through the firewall rules to see if I can narrow it down.

**Comment by dperson** [comment]
2 quick options:
* if you don't want the firewall blocking things you can disable it (don't set `-f` or the `FIREWALL` environment variable and remove the `/vpn/.firewall*` file(s))
* I can try to help you troubleshoot the problem, if you give me more details about the port forward, any error messages that you're getting in `docker logs <container_name>`, etc

**Comment by fad6b5f049** [comment]
No errors since I'm pretty sure the image works as intended (great image btw).

Preferably I don't want to disable the firewall, just allow the specific incoming port through it whilst the VPN is connected. That being said, kill-switch should still disable everything in order to not leak the address on a disconnect of the connection.

The way this specific (and a few others + self-hosted) VPN provider works is that you get assigned a/several specific port(s) that you can utilize for port-forwarding thus making one able to run hosting-services through the VPN provider.

The process is basically: You ask for a open-port from the VPN provider, they give you port (example) 11412, you then configure your service to listen for connections on port 11412 (I saw earlier that you mentioned using PIA, for PIA you can configure getting a random open-port, this works similarly although the assigned port is pre-defined so you don't have to request it upon connection).

Basically my initial question was thus - is there an environmental variable or a command that exists to allow these/this port(s) through the firewall at creation of the container? (I'm assuming no).

**Comment by dperson** [comment]
Yes, it's covered in the other issue #201 that you referenced.

**Comment by fad6b5f049** [comment]
Like dperson said, it is covered in issue #201.

I tried several times to get it to work, and couldn't understand why it didn't. Turns out, after leaving the container running for a while it suddenly started working even though it didn't at the start of the container. So I guess it works, just don't be impatient like me!

Here's what I use to run the container so that people with the same "issue" (not really an issue) can have a peek:

```
docker run -d \
--name=openvpn-client \
--restart=unless-stopped \
--cap-add=NET_ADMIN \
--net=$dockernetwork \
--device /dev/net/tun \
--sysctl net.ipv6.conf.all.disable_ipv6=0 \
-v $containerlocation/dperson/openvpn-client/vpn:/vpn \
-e DNS=<VPN DNS> \
-e FIREWALL="" \
-e VPNPORT=<PORT TO BE FORWARDED> \
-e ROUTE='<HOME NETWORK>' \
dperson/openvpn-client:latest
```
